scripts/missions/cop/3_3_The_Road_Forks.lua

```diff
@@ -0,0 +1,465 @@
+-----------------------------------
+-- The Road Forks
+-- Promathia 3-3
+-----------------------------------
+-- !addmission 6 325
+
+-- San d'Oria Path
+-- Arnau      : !pos 148 0 139 231
+-- Chasalvige : !pos 96.432 -0.520 134.046 231
+-- Guilloud   : !pos -123.770 -6.654 -469.062 2
+-- Hinaree    : !pos -301.535 -10.199 97.698 230
+
+-- Windurst Path
+-- Ohbiru-Dohbiru    : !pos 23 -5 -193 238
+-- Yoran-Oran        : !pos -109.987 -14 203.338 239
+-- Kyume-Romeh       : !pos -58 -4 23 238
+-- Honoi-Gumoi       : !pos -195 -11 -120 238
+-- Loose Sand        : !pos 478.8 20 41.7 7
+-- Cradle of Rebirth : !pos 320 -23 -15.9 7
+-- Yujuju            : !pos 201.523 -4.785 138.978 240
+-- Tosuka-Porika     : !pos -26 -6 103 238
+
+-- Completion
+-- Cid : !pos -12 -12 1 237
+-----------------------------------
+-- NOTE: This mission uses extended Mission Status.  See documentation for CoP MissionStatus for
+-- predefined values.  These must not be changed!
+-----------------------------------
+require('scripts/globals/interaction/mission')
+require('scripts/globals/keyitems')
+require('scripts/globals/missions')
+require('scripts/globals/npc_util')
+require('scripts/globals/zone')
+-----------------------------------
+local attohwaChasmID      = require('scripts/zones/Attohwa_Chasm/IDs')
+local carpentersLandingID = require('scripts/zones/Carpenters_Landing/IDs')
+-----------------------------------
+
+local mission = Mission:new(xi.mission.log_id.COP, xi.mission.id.cop.THE_ROAD_FORKS)
+
+mission.reward =
+{
+    nextMission = { xi.mission.log_id.COP, xi.mission.id.cop.TENDING_AGED_WOUNDS },
+}
+
+-- A 30 minute timer is started once obtaining the Mimeo Jewel for Windurst Path.  Create a
+-- looping timer to monitor this, supply periodic messages, and check for breaking conditions
+local jewelTimer
+jewelTimer = function(player)
+    if
+        not player or
+        not player:hasKeyItem(xi.ki.MIMEO_JEWEL)
+    then
+        return
+    end
+
+    local secondsRemaining = mission:getLocalVar(player, 'Timer') - os.time()
+    if secondsRemaining <= 0 then
+        -- There are some conditions that can instantly break the jewel.  Short-circuit out
+        -- here and display the final message should that occur.
+        player:messageSpecial(attohwaChasmID.text.MIMEO_JEWEL_OFFSET + 4)
+        player:delKeyItem(xi.ki.MIMEO_JEWEL)
+    else
+        local messageOffset = mission:getLocalVar(player, 'Option')
+        local nextMessageTime = 30 - (messageOffset * 6)
+        local minutesRemaining = secondsRemaining / 60
+
+        if minutesRemaining <= nextMessageTime then
+            player:messageSpecial(attohwaChasmID.text.MIMEO_JEWEL_OFFSET + messageOffset - 1) -- TODO: Might need KI Arg
```

Comment:
You could store this on the player as a localvar?

Reply:
messageOffset is stored as a localVar.  It changes on each message update, though we're not 0-indexed for the messages displayed for safety (there's a lot of duplicates for the first message text, so its 1-indexed, with the first message being 1 lower).